client/src/store/utils/thunkCreators.js

```diff
@@ -67,13 +68,29 @@ export const logout = (id) => async (dispatch) => {
   }
 };
 
+const findLastReadMessageId = (messages, userId) => {
+  let lastReadMessageId = -1;
+  messages.forEach(message=>{
+    if (message.id > lastReadMessageId && message.read === true && message.senderId === userId)
+      {
+        lastReadMessageId = message.id
+      }
+  })
+  return lastReadMessageId === -1 ? undefined: lastReadMessageId;
+};
+
 // CONVERSATIONS THUNK CREATORS
 
-export const fetchConversations = () => async (dispatch) => {
+export const fetchConversations = (userId) => async (dispatch) => {
   try {
     const { data } = await axios.get("/api/conversations");
     data.forEach(conversation => 
-      {conversation.messages.sort((a, b) => a.createdAt.localeCompare(b.createdAt));})
+      {
+        conversation.messages.sort((a, b) => a.createdAt.localeCompare(b.createdAt));
+        conversation.lastReadMessageId = findLastReadMessageId (conversation.messages, userId);
+        conversation.unreadMessagesCount = conversation.messages.filter(message => message.read === false 
+          && message.senderId === conversation.otherUser.id).length;
```

Comment:
Notice the repetition on counting unread messages... It would be better to retrieve this value directly in the backend. Similarly, the last message ID could be retrieved in the backend as well.

Reply:
Yes I agree. I acknowledge that in my loom video message too. But for the purposes of this task, I am keeping it as it is for now